ci/make_container_images.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# This script is intended to be used by Cirrus-CI, from the VM
+# built by the 'image_builder' makefile target in this repo.  Use
+# of this script in any other context/environment is unlikely to
+# function as intended.
+
+set -e
+
+SCRIPT_FILEPATH=$(realpath "$0")
+SCRIPT_DIRPATH=$(dirname "$SCRIPT_FILEPATH")
+REPO_DIRPATH=$(realpath "$SCRIPT_DIRPATH/../")
+
+# shellcheck source=./lib.sh
+source "$REPO_DIRPATH/lib.sh"
+
+# When running under Cirrus-CI on a branch, also tag/push "latest" images
+PUSH_LATEST=0
+
+# shellcheck disable=SC2154
+if [[ -z "$CI" ]] || [[ "$CI" != "true" ]] || [[ "$CIRRUS_CI" != "$CI" ]]; then
+    die "Unexpected \$CI=$CI and/or \$CIRRUS_CI=$CIRRUS_CI"
+elif [[ -z "$IMG_SFX" ]]; then
+    die "Required non-empty values for \$IMG_SFX=$IMG_SFX"
+elif [[ -z "$TARGET_NAME" ]]; then
+    die "Required makefile target \$TARGET_NAME to be non-empty"
+elif [[ -z "$DEST_FQIN" ]]; then
+    die "Required destination FQIN \$DEST_FQIN to be non-empty"
+# avoid exposing secret contents
+elif [[ "${#LOGIN_CMD}" -le 0 ]] || grep -Eq "ENCRYPTED" <<<"$LOGIN_CMD"; then
```

Comment:
This is really hard to understand: under what circumstances will a string length be less than zero? Why are you using `grep -E` on something that is in no way a regex (and could better be handled by fgrep)? And, this isn't running under `set -x`, so where is the risk of exposure? Couldn't this be handled via:
```bash
elif [[ $LOGIN_CMD =~ ENCRYPTED ]]; then
    die "\$LOGIN_CMD did not get decrypted by Cirrus"
elif [[ ${#LOGIN_CMD} -le 5 ]]; then
    die "\$LOGIN_CMD is a ${#LOGIN_CMD}-length string; something almost certainly went wrong with Cirrus decryption"
fi
```

Reply:
> so where is the risk of exposure

Myself, I've done this:  Debugging and do a `set -x` on the whole script then run it through CI :confounded: 

Otherwise your suggestion is sound/good idea.